### ISSUE TITLE
pytest -k value incorrectly triggers 'Path not found' warning

### ISSUE THREAD

**Issue by MarcusJellinghaus** [MarcusJellinghaus/mcp-code-checker#115]
## Description

`sanitize_extra_args()` in `src/mcp_tools_py/code_checker_pytest/utils.py` treats all non-flag arguments as potential file paths. When a flag like `-k` takes a value argument (e.g. `-k order`), the value is incorrectly checked as a file path, producing a spurious warning:

```
Path 'order' not found relative to project_dir.
```

## Root Cause

The argument parser processes args sequentially without understanding flag-value pairs. When it sees `-k order`:

1. `-k` is recognized as a flag and skipped
2. `order` falls through to path validation on the next iteration
3. No file named `order` exists, so the warning is emitted

## Affected Flags

Any pytest flag that takes a next-arg value: `-k`, `-m`, `-p`, `--timeout`, etc.

## Expected Behavior

Values belonging to flags like `-k` and `-m` should be consumed together with their flag and not treated as file paths.

## Location

- `src/mcp_tools_py/code_checker_pytest/utils.py` — `sanitize_extra_args()`, lines ~62-94
- Tests: `tests/test_code_checker_pytest/test_extra_args.py`

**Comment by MarcusJellinghaus** [comment]
Duplicate of #145 — same root cause in `sanitize_extra_args()`. Consolidating into #145, which has the most complete reproduction and fix proposals. The code-location info from this issue has been ported into #145.
